File: app/domain/category.py
from dataclasses import dataclass
from typing import override


@dataclass
class Category:
    id: int | None = None
    title: str = ""
    alias: str = ""
    template: str | None = None
    page: bool | None = False

    @override
    def __str__(self) -> str:
        return f"Category(id={self.id}, title={self.title})"

File: rust/crates/domain/src/category.rs
use serde::{Deserialize, Serialize};

/// Blog category entity.
#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct Category {
    pub id: Option<i32>,
    pub title: String,
    pub alias: String,
    pub template: Option<String>,
    pub page: Option<bool>,
}
